### DISCUSSION TITLE
Offline speech recognition not available on device

### DISCUSSION THREAD

**Discussion by sej69** [CommunityToolkit/Maui#1680] [Q&A]
In @VladislavAntonyuk Microsoft blog post here:

[https://vladislavantonyuk.github.io/articles/Speech-recognition-with-.NET-MAUI/](url)

It's stated:

"The availability of an Internet connection may be necessary depending on the chosen recognition language. On Windows, the speech recognition system automatically adjusts between online and offline modes based on Internet accessibility."

I'm using US English, and the speech recognizer is not coming online if I'm not connected to the internet.  My need is to do on device speech recognition.  The moment I turn internet on using my development iPAD it does work.

Anyone know the specifics around what's needed for on device?

**Comment by VladislavAntonyuk** [comment]
Speech recognition is limited to the latest iPads. And in the latest version we removed offline speech recognition for Windows.

**Comment by sej69** [comment]
Is that include the latest iPhone as well?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by sej69** [comment]
Also, can you provide the minimum version I can code into the app?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by sej69** [comment]
@VladislavAntonyuk I went out and purchased a gen 10 ipad yesterday and spent all afternoon updating and getting it into my dev environment.  There is a noticeable difference in performance on this version compared to the gen 4 ipad mini.  I'm going to try my test project out this morning to see if the recognition complete flag gets set now.

Does this library work ok with iPhones as well?  I don't have an iPhone to test.  Also, can you provide the exact version of hardware I need to code into my app for support?  I want to make sure that user's downloading this app have the appropriate hardware level. Thanks!

**Reply by sej69** [reply]
RecognitionComplete is still not working so I'm stuck with using my timing routines.

**Comment by CliffAgius** [comment]
Closed as answered, if closed in error please let us know to re-open.
